### PR TITLE
network: cleanup link state file

### PR DESCRIPTION
**Pull request by yuwata**
Replaces #16209.

cc @ssahani.

### CODE REVIEW

File: src/libsystemd/sd-network/sd-network.c [hunk 1]
```diff
@@ -236,14 +232,6 @@ _public_ int sd_network_link_get_dnssec_negative_trust_anchors(int ifindex, char
...
 
-_public_ int sd_network_link_get_timezone(int ifindex, char **ret) {
-        return network_link_get_string(ifindex, "TIMEZONE", ret);
-}
-
-_public_ int sd_network_link_get_dhcp4_address(int ifindex, char **ret) {
-        return network_link_get_string(ifindex, "DHCP4_ADDRESS", ret);
-}
-
 _public_ int sd_network_link_get_dns(int ifindex, char ***ret) {
```
Comment (ssahani) [inline src/libsystemd/sd-network/sd-network.c:235]: Should we remove this. This belongs to released version. API breakage.

File: src/network/networkctl.c [hunk 2]
```diff
@@ -958,12 +959,18 @@ static int dump_addresses(
...
 
-                        p = pretty;
-                        pretty = strjoin(pretty , " (DHCP4)");
-                        if (!pretty)
+                        r = sd_dhcp_lease_get_server_identifier(lease, &server_address);
+                        if (r >= 0)
+                                p = strjoin(pretty, " (DHCP4 via ", inet_ntop(AF_INET, &server_address, s, sizeof(s)), ")");
+                        else
+                                p = strjoin(pretty, " (DHCP4)");
+                        if (!p)
```
Comment (ssahani) [inline src/network/networkctl.c:970]: I think this situation will never happen. Without DHCP server client wont get a IP. But I think OK.

File: src/libsystemd/sd-network/sd-network.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd/sd-network/sd-network.c:235]: sd-network is internal API. Not exposed yet. So, we can safely remove this.

File: src/network/networkctl.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkctl.c:970]: But, reading lease file may fail. Just for safety.

Comment (ssahani) [reply src/network/networkctl.c:970]: Unless someone deletes the lease between which is highly unlikely. But ok.

**Comment by ssahani** [comment]
LGTM

File: src/network/networkctl.c [hunk 3]
```diff
@@ -958,12 +959,18 @@ static int dump_addresses(
...
-                        _cleanup_free_ char *p = NULL;
+                        struct in_addr server_address;
+                        char *p, s[INET_ADDRSTRLEN];
 
-                        p = pretty;
-                        pretty = strjoin(pretty , " (DHCP4)");
-                        if (!pretty)
+                        r = sd_dhcp_lease_get_server_identifier(lease, &server_address);
+                        if (r >= 0)
+                                p = strjoin(pretty, " (DHCP4 via ", inet_ntop(AF_INET, &server_address, s, sizeof(s)), ")");
```
Comment (keszybz) [inline src/network/networkctl.c]: Please wrap `inet_ntop()` in `strna()`. inet_ntop may fail.

Comment (keszybz) [reply src/network/networkctl.c]: Or maybe even do this:
```C
r = sd_dhcp_lease_get_server_identifier(lease, &server_address);
if (r >= 0 && inet_ntop(AF_INET, &server_address, s, sizeof s))
     p = strjoin(pretty, " (DHCP4 via ", s, ")");
else
     p = strjoin(pretty, " (DHCP4)");
```

**Comment by yuwata** [comment]
@keszybz Thank you for the review. Addressed the above point. PTAL.

**Pull request merged**